scanomatic/image_analysis/signal.py

```diff
@@ -637,7 +639,7 @@ def get_closest_signal_pair(
 
     S = np.array(signals)
     if S.size == 0:
-        return None, None
+        raise SignalError("No signal pairs found")
```

Comment:
The None tuple wasn't handled anywhere

Reply:
sounds good consider getting rid of the `S` variable as something like `signals_arr` or something too.